testing/python/collect.py

```diff
@@ -1076,3 +1076,19 @@ def test_real():
         'WC2 *',
         '*1 passed, 1 pytest-warnings in *',
     ])
+
+
+def test_class_injection_does_not_break_collection(testdir):
+    testdir.makeconftest("""
+        from test_inject import TestClass
+        def pytest_generate_tests(metafunc):
+            TestClass.changed_var = {}
+    """)
+    testdir.makepyfile(test_inject='''
+         class TestClass(object):
+            def test_injection(self):
+                """Test being parametrized."""
+                pass
+    ''')
+    result = testdir.runpytest()
+    assert "RuntimeError: dictionary changed size during iteration" not in result.stdout.str()
```

Comment:
made this a bit more general so that it works with pytest-sugar as well as normal pytest outputs.